components/core/src/clp_s/search/NarrowTypes.cpp

```diff
@@ -67,6 +84,50 @@ std::shared_ptr<Expression> NarrowTypes::narrow(std::shared_ptr<Expression> cur)
         if (false == column->matches_any(cAllTypes)) {
             return EmptyExpr::create();
         }
+
+        /**
+         * Fix for bug #254
+         * If a filtering operation which can originally match a clp string and a var string is
+         * narrowed to just one of those underlying types AND the filter is performing a not equals
+         * operation then we need to replace the filter with an OR across that filter expression,
+         * and an EXISTS expression for the other underlying type.
+         *
+         * This is because if we are trying to see if some value is not equal to a specific variable
+         * string we can guarantee that every clp string is not equal to that specific variable
+         * string (because that variable string must not contain a space, but any clp string must).
+         * The reverse is true for checking if some value matches a specific clp string.
```

Comment:
```suggestion
         * Fix for bug y-scope/clp#254.
         * If:
         * - a filtering operation that can match a clp string and a var string is narrowed to just
         *   one of those underlying types
         * - AND the filter is performing a not equals operation
         * then we need to replace the filter with an OR across:
         * - the filter expression
         * - and an EXISTS expression for the other underlying type.
         *
         * This is because if we are trying to determine if some value is not equal to a specific
         * variable string, we can guarantee that every clp string is not equal to that specific
         * variable string (because that variable string must not contain a space, but any clp
         * string must). The reverse is true for checking if some value matches a specific clp
         * string.
```